### ISSUE TITLE
Clarify the behavior of file descriptors when calling `exec*`

### ISSUE THREAD

**Issue by Yoric** [redox-os/redox#792]
In general, we probably want to close file descriptors.

I suspect that, to simplify IPC and capabilities, we may want to support a variant of `exec*` that lets us pass file descriptors, though.

**Comment by jackpot51** [comment]
This is the change I like the most. If all file descriptors where by default CLOEXEC, or if they had to be specified directly in an argument to exec, I think it would be much safer

**Comment by Yoric** [comment]
Yes, that's what I had in mind. I haven't come up with a good API for exec-with-file-descriptors yet, but that shouldn't be too hard.
